core/debugger/remote_debugger.cpp

```diff
@@ -599,8 +598,7 @@ void RemoteDebugger::poll_events(bool p_is_idle) {
 			reload_all_scripts = false;
 		} else if (!script_paths_to_reload.is_empty()) {
 			Array scripts_to_reload;
-			for (int i = 0; i < script_paths_to_reload.size(); ++i) {
-				String path = script_paths_to_reload[i];
+			for (const String path : script_paths_to_reload) {
```

Comment:
Pardon my ignorance - elsewhere I'm seeing similar loops but with the variable being a reference, eg. this would be:

```
for (const String &path : script_paths_to_reload) {
```

Is there any material difference between the two forms, or is it a preference? Perhaps we could use one style for consistency?

Reply:
> Some cases are without reference, they are intended partially as they do conversion from Variant, anything that parses an Array and doesn't specifically use Variant does this, for example

Does this apply?

Reply:
This is exactly why, explained in the OP :)

Reply:
Right, my bad, thanks for confirming :)

Reply:
No worries! The reason is you can't bind a reference to a cast

Reply:
Would it make sense to add a comment in such cases?
The risk is very high that future refactorings optimize `String` to `const String&` otherwise.

Reply:
No risk, it'll fail compilation